spec/rspec/mocks/matchers/have_received_spec.rb

```diff
@@ -459,6 +459,73 @@ module Mocks
             }.to raise_error(/received :two out of order/m)
           end
 
+          context "Specifying ordering for calls that are interleaved with each other" do
+            it 'passes when using `have_received`' do
+              dbl = spy
+
+              dbl.one
+              dbl.one
+              dbl.two
+              dbl.one
+              dbl.two
+
+              expect(dbl).to have_received(:one).twice.ordered
+              expect(dbl).to have_received(:two).once.ordered
+              expect(dbl).to have_received(:one).once.ordered
+              expect(dbl).to have_received(:two).once.ordered
+            end
+
+            it 'fails when using `have_received` with out of order interleaved messages' do
+              dbl = spy
+
+              dbl.one
+              dbl.one
+              dbl.two
+              dbl.one
+              dbl.two
+
+              expect {
+                expect(dbl).to have_received(:two).once.ordered
+                expect(dbl).to have_received(:one).twice.ordered
+                expect(dbl).to have_received(:one).once.ordered
+                expect(dbl).to have_received(:two).once.ordered
+              }.to raise_error(/received :one out of order/m)
```

Comment:
Can this expectation be more specific about which of the ordering expectations raises the error?

(Also, this can use `fail_with`, rather than `raise_error`, I believe, which specifies the exception type as well).